doc: enabling settings pool on vttablet (#1145)
vitessio/website: content/en/docs/15.0/reference/query-serving/reserved-conn.md

result in reserved connections being employed for some/all of the application's
sessions.

### Settings pool and reserved connections

> There are multiple sections below that highlight how reserved connections get triggered for different use cases.

Reviewer: ```suggestion
We will see how reserved connections get triggered for different use cases in subsequent sections of this document.
```

> There is an issue when a reserved connection is used as it turns off the reuse of connection in the vttablet. More details about it are given [below](#number-of-vttablet---mysql-connections).

Reviewer: ```suggestion
What we want to highlight at the beginning is that there is a known issue when a reserved connection is used as it cannot be reused by vttablet. More details about it are given [below](#number-of-vttablet---mysql-connections).
```

> The current pool implementation is enhanced to keep the connections with settings in the pool and not to pin the connection to the session.

Reviewer: ```suggestion
To solve this problem, the connection pool implementation used by vttablet has been enhanced to keep the connections with settings in the pool and not to pin the connection to the client session.
```

> This will avoid the pitfall of reserved connection i.e. blocking MySQL connection which leads to MySQL running out of connections.

Reviewer: ```suggestion
With this enhancement, we reduce the likelihood of MySQL running out of connections due to reserved connections, because the scenarios where we still need reserved connections are sharply reduced.
```

> This also helps in retaining the other benefits of having a connection pool.

Reviewer: ```suggestion
```

> This feature is currently disabled by default and can be enabled with a flag `--queryserver-enable-settings-pool` on vttablet.

Reviewer: ```suggestion
Since this is a new feature, it is currently disabled by default. It can be enabled by setting the flag `--queryserver-enable-settings-pool` on vttablet.
```

This only works for the cases when the system variables need a reserved connection.

> There are still cases like [temporary tables](#temporary-tables-and-reserved-connections) and [advisory locks](#get_lock-and-reserved-connections) where the reserved connection will continue to be used.

Reviewer: ```suggestion
This change takes effect only for the cases when system variable changes need a reserved connection.
There are still cases like [temporary tables](#temporary-tables-and-reserved-connections) and [advisory locks](#get_lock-and-reserved-connections) where reserved connections will continue to be used.
```
